include/aie/Dialect/AIEX/Transforms/AIEXPasses.td

```diff
@@ -147,4 +147,37 @@ def AIEDmaToNpu : Pass<"aie-dma-to-npu", "AIE::DeviceOp"> {
   ];
 }
 
+def AIEMaterializeBDChains : Pass<"aie-materialize-bd-chains", "AIE::DeviceOp"> {
+  let summary = "Concretize aie.bd_chain ops at aiex.start_task use sites";
+  let description = [{
+    Turns pre-defined BD chains with parameters into concrete BD configuration instructions without parameters at their usage sites.
+  }];
+
+  let constructor = "xilinx::AIEX::createAIEMaterializeBDChainsPass()";
+  let dependentDialects = [
+    "xilinx::AIE::AIEDialect",
+    "xilinx::AIEX::AIEXDialect",
+  ];
+}
+
+def AIEAssignRuntimeSequenceBDIDs : Pass<"aie-assign-runtime-sequence-bd-ids", "AIE::DeviceOp"> {
+  let summary = "Assign IDs to Buffer Descriptors Configured in the Runtime Sequence";
+
+  let constructor = "xilinx::AIEX::createAIEAssignRuntimeSequenceBDIDsPass()";
+  let dependentDialects = [
+    "xilinx::AIE::AIEDialect",
+    "xilinx::AIEX::AIEXDialect",
+  ];
+}
```

Comment:
Currently it's not combined because `aie-assign-bd-ids` is an AIE pass, whereas this is an AIEX pass operating on AIEX instructions.

What is different is that this "allocator" allows re-use of BD IDs, making it slightly more complicated. The other one sequentially assigns IDs until it runs out. Other than that they do the same thing, so could be merged if AIEX stuff makes it into AIE.